### ISSUE TITLE
activation: bind post data to genesis id

### ISSUE THREAD

**Issue by dshulyak** [spacemeshos/go-spacemesh#3507]
part of: https://github.com/spacemeshos/pm/issues/135
depends on: https://github.com/spacemeshos/go-spacemesh/issues/3505

this can be implemented independently from https://github.com/spacemeshos/pm/issues/168

- initialize PostSetupManager with genesis id. when `imgr.id` is passed to Initiliazer and Prover prefix it with genesis id
- initialize Validator with genesis id. prefix miner id with genesis id when ValidatePost is called 

@moshababo could you please review this task?

**Comment by moshababo** [comment]
Currently, [gpu-post](https://github.com/spacemeshos/gpu-post) doesn't support `id` size other than 32 bytes. So we need to change that, or let `id` be the hash of genesis ID and miner ID. 

Besides that, LGTM.

**Comment by lrettig** [comment]
My understanding was that POPS-VRF (https://github.com/spacemeshos/pm/issues/168) would achieve this. @dshulyak is the idea here that this is a stopgap until that is done? Or do you think we need both for some reason?

**Comment by dshulyak** [comment]
the way pops-vrf described in https://github.com/spacemeshos/pm/issues/168 allows to generate post data in advance. i don't recall the motivation, but i asked on the last call if we can reuse post data between networks and the answer was no.

**Comment by moshababo** [comment]
Closing in favor of https://github.com/spacemeshos/pm/issues/172.
